src/utils/loadable.ts

```diff
@@ -0,0 +1,67 @@
+import { atom } from 'jotai'
+import type { Atom } from 'jotai'
+import { getWeakCacheItem, setWeakCacheItem } from './weakCache'
+
+const loadableAtomCache = new WeakMap()
+
+export type Loadable<Value> =
+  | LoadableLoading
+  | LoadableError
+  | LoadableValue<Value>
```

Comment:
We prefer simpler (highly subjective though) type aliases, and avoid exporting types unless it's a huge pain. Let's see how it goes without exporting it.
```suggestion
type Loadable<Value> =
  | { state: 'loading' }
  | { state: 'hasError'; error: unknown }
  | { state: 'hasData'; data: Value }
```

Reply:
Will do